luaui/Widgets/cmd_blueprint.lua

```diff
@@ -411,9 +477,16 @@ local function deleteBlueprint(index)
 
 	if #blueprints == 0 then
 		setSelectedBlueprintIndex(nil)
-	elseif selectedBlueprintIndex > #blueprints then
-		setSelectedBlueprintIndex(selectedBlueprintIndex - 1)
+	elseif index > selectedBlueprintIndex then
```

Comment:
Right now as far as I can see, the only call to deleteBlueprint is `deleteBlueprint(selectedBlueprintIndex)`.  While this code looks ok for robustness, looks a bit wasteful. Maybe tests could do this, but you also deleted that test.

For code simplicity I think best would be to just check current index but change deleteBlueprint(index) to deleteSelectedBlueprint() that way you can be sure nobody will call deleteBlueprint with a different index than the selected one and have just two conditions #blueprints == 0 and else just go next or prev as you feel better.

Anyways if you really want to keep it like this I guess it's ok too. It's not a big deal, it's just dead code atm but it all depends in your future plans.